### ISSUE TITLE
Icons lost whel transforming html to pdf

### ISSUE THREAD

**Issue by HaojunRen** [wkhtmltopdf/wkhtmltopdf#4815]
**wkhtmltopdf version(s) affected**:
0.12.6-1

**OS information**
Window 10 Prefessional

**Description**
Icons lost while transforming html to pdf

**How to reproduce**
http://nepxion.gitee.io/docs/web-doc/Polaris%E3%80%90%E5%8C%97%E6%9E%81%E6%98%9F%E3%80%91%E4%BC%81%E4%B8%9A%E7%BA%A7%E4%BA%91%E5%8E%9F%E7%94%9F%E5%BE%AE%E6%9C%8D%E5%8A%A1%E6%A1%86%E6%9E%B6.html

transform the html file above to pdf

the first picture is for html
![图片](https://user-images.githubusercontent.com/42354711/91954403-56e2ae00-ed34-11ea-942a-d5eb8ce207b8.png)

the second picture is for pdf
![图片](https://user-images.githubusercontent.com/42354711/91954533-81cd0200-ed34-11ea-8822-00d7ef3204c1.png)

**Comment by PhilterPaper** [comment]
Are these emojis or images (files)? Take a look at #2913 -- maybe it's a similar problem? I'm not sure if any fix was come up with for 2913. If your HTML input is using any fancy CSS or JS to display the icons, it's possible that wkHTMLtoPDF's WebKit is not able to display it.

**Comment by HaojunRen** [comment]
Sure, they are maybe emojis those are from https://shields.io/. But in some pdf tools, they can display correctly

**Comment by PhilterPaper** [comment]
What exactly do you mean by "in some pdf tools, they can display correctly"? That the PDF produced by wkHTMLtoPDF displays correctly in some (but not all) PDF readers, _or_ some other HTML-to-PDF converters produce a good PDF?

Looking at the source, I see this is an `<img>` (in `<a>`) that says `type="image/svg+xml;charset=utf-8" src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3...`. wkHTMLtoPDF is known not to handle SVG very well, so my guess is that the icon part of the image is drawn in SVG and can't be rendered here. If there is a version that is a pure PNG or other _raster_ image format, you might have better luck with that.

**Comment by HaojunRen** [comment]
Would you please provide a simple feature:
if it can not render svg, that place does not display an empty icon. That means if svg is 100x100 size, pdf DONT show 100x100 empty image, it is better to show nothing there, it looks comfortable

**Comment by PhilterPaper** [comment]
You might want to take a look at #4808 -- that one seems to be reporting success with SVG+XML.

**Comment by HaojunRen** [comment]
Yeah, the solution you provided can work. But it is so difficult for me with so many steps:
1. conver md to html
2. open html to rewrite svg codes part
3. convert to pdf

As i suggested 5 days ago, if wkh met the invalid svg, DON'T add SPACES there, just as follows
![图片](https://user-images.githubusercontent.com/42354711/92392692-3c07a380-f151-11ea-93ff-5687384260ac.png)

I rather like those icons display in html, and NOT display in pdf. Would you please? Thanks a lot!
